mne_bids/utils.py

```diff
@@ -88,10 +88,14 @@ def _handle_kind(raw):
     """Get kind."""
     if 'meg' in raw:
         kind = 'meg'
-    elif 'eeg' in raw:
-        kind = 'eeg'
-    elif 'ecog' in raw:
+    elif 'ecog' in raw and 'eeg' not in raw:
         kind = 'ieeg'
+    elif 'eeg' in raw and 'ecog' not in raw:
```

Comment:
you don't need the second part of this `if` clause if you raise the `ValueError` before these `if`/`elif` clauses. In general I find it much easier to read code and write the logic of `if`/`else` statements if the errors are raised first. It excludes many cases for you

Reply:
+1